src/CustomiesListener.php

```diff
@@ -36,16 +34,10 @@ public function __construct() {
 
 	public function onDataPacketSend(DataPacketSendEvent $event): void {
 		foreach($event->getPackets() as $packet){
-			if($packet instanceof BiomeDefinitionListPacket) {
-				// ItemComponentPacket needs to be sent after the BiomeDefinitionListPacket.
-				if($this->cachedItemComponentPacket === null) {
-					// Wait for the data to be needed before it is actually cached. Allows for all blocks and items to be
-					// registered before they are cached for the rest of the runtime.
-					$this->cachedItemComponentPacket = ItemComponentPacket::create(CustomiesItemFactory::getInstance()->getItemComponentEntries());
-				}
-				foreach($event->getTargets() as $session){
-					$session->sendDataPacket($this->cachedItemComponentPacket);
-				}
+			if($packet instanceof ItemRegistryPacket) {
+				(function() : void{
+					$this->entries = array_merge($this->entries, CustomiesItemFactory::getInstance()->getItemTableEntries());
```

Comment:
Tbh I'd like to just set entries into ItemTypeDictionary using ReflectionClass instead of merging here - probably very very costly to do this here

Reply:
Any reason not to do this?